### ISSUE TITLE
CH11 MC Dropout question

### ISSUE THREAD

**Issue by ycbq999** [ageron/handson-ml2#70]
I may not quite understand the process of MC dropout. 
When you force the dropout on during the training, then is it just like the normal neural network?

**Comment by ageron** [comment]
Hi @ycbq999 ,
Thanks for your question! Regular dropout only happens during training: at each training iteration, we randomly set the output of some neurons to zero, with probability `dropout_rate`, and we multiply all the weights by `1 / dropout_rate`. This pushes the network to be robust to noise, making it better at generalizing to new data (it's a regularization technique). But once training is finished, regular dropout does not do anything anymore: the neural network is used normally, without turning off neurons randomly or tweaking the weights.

With MC dropout, training is exactly the same as for regular dropout, but we continue to apply dropout after training, when making predictions. As a result, the neural network will make slightly different predictions when you run it multiply times on the same input data. For example, if you trained it on the MNIST dataset and you ask it to make a prediction for an image representing the digit 5, it might output something like [0.,  0., 0., 0.1, 0.0, 0.9, 0., 0., 0., 0.] the first time you ask, and [0.,  0., 0., 0.14, 0.0, 0.85, 0., 0., 0.01, 0.] the second time you ask (in other words, the first time it says that the probability that the image represents a 5 is 90%, and the proba that it represents a 3 is 10%, but the second time you ask it to classify the same image, it says there's an 85% chance that it's a 5, and 14% chance that it's a 3, and 1% chance that it's an 8).
This randomness is useful because if you ask the neural network to classify the image 100 times, it will give you a more precise idea of what the real probabilities are, and what the real uncertainty is.
However it comes at a cost since it requires running the neural net multiple times.

One great thing about MC dropout is that it does not require retraining existing models that were trained using regular dropout.

Hope this helps!

**Comment by ageron** [comment]
Is it clear now? May I close this issue?
